packages/client/src/WalletAPIClient.ts

```diff
@@ -82,9 +85,25 @@ export class WalletAPIClient extends RpcNode<
    */
   public exchange: ExchangeModule;
 
+  /**
+   * Instance of the Custom module
+   */
+  public custom: CustomGetter extends (client: WalletAPIClient<M>) => infer U
```

Comment:
if we define

```
type CustomPropertyType<M extends CustomModule, CustomGetter extends CustomGetterType<M>> = 
  CustomGetter extends (client: WalletAPIClient<M>) => infer U ? U : CustomModule | Record<string, CustomModule>;
```
right after the `CustomGetterType` and before the class definition, then we can replace the custom property type definition as follow:

```
public custom: CustomPropertyType<M, CustomGetter>;
```

Reply:
Unfortunately you cannot define this type outside the class as it contains generics